plugins/org.eclipse.sirius.diagram.sequence/src/org/eclipse/sirius/diagram/sequence/business/internal/layout/vertical/SequenceVerticalLayout.java

```diff
@@ -771,45 +771,38 @@ private boolean layoutUnconnectedLostMessageEnd() {
     private boolean layoutInteractionContainer() {
         boolean applied = false;
         Set<InteractionContainer> allInteractionContainers = this.sequenceDiagram.getAllInteractionContainers();
-        if (!allInteractionContainers.isEmpty() && getInteractionContainerBounds() != null) {
+        if (!allInteractionContainers.isEmpty()) {
             // Reset height of the interaction container
-            Bounds interactionContainerBounds = getInteractionContainerBounds();
-            interactionContainerBounds.setHeight(InteractionContainer.DEFAULT_HEIGHT);
-            for (InstanceRole instanceRole : this.sequenceDiagram.getAllInstanceRoles()) {
-                // Gather instance role bounds
-                final Node instanceRoleNode = instanceRole.getNotationNode();
-                LayoutConstraint layoutConstraint = instanceRoleNode.getLayoutConstraint();
-                // Check that the lifeline is within bounds of the interaction container, resize it otherwise
-                if (layoutConstraint instanceof Bounds instanceRoleBounds && instanceRole.getLifeline().some()) {
-                    Lifeline lifeline = instanceRole.getLifeline().get();
-                    Node lifelineNode = lifeline.getNotationNode();
-                    // lifeline position is relative to the parent instance role position
-                    if (lifelineNode.getLayoutConstraint() instanceof Bounds lifelineBounds
-                            && instanceRoleBounds.getY() + lifelineBounds.getY() + lifelineBounds.getHeight() + InteractionContainer.MARGIN > interactionContainerBounds.getHeight()) {
-                        interactionContainerBounds.setHeight(instanceRoleBounds.getY() + lifelineBounds.getY() + lifelineBounds.getHeight() + InteractionContainer.MARGIN);
+            InteractionContainer interactionContainer = allInteractionContainers.iterator().next();
+            Node node = interactionContainer.getNotationNode();
+            LayoutConstraint interactionContainerLayoutConstraint = node.getLayoutConstraint();
+            if (interactionContainerLayoutConstraint instanceof Bounds interactionContainerBounds) {
+                Rectangle rectangle = new Rectangle(0, 0, InteractionContainer.DEFAULT_WIDTH, InteractionContainer.DEFAULT_HEIGHT);
+                for (InstanceRole instanceRole : this.sequenceDiagram.getAllInstanceRoles()) {
```

Comment:
Check if you can use the lifeline ranges from the finalRanges parameter of applyComputedLayout method. 
It has been used to setLifeline vertical range (and InstanceRole layout, see Lifeline.setVerticalRange)